### PR TITLE
Redux: Batch update of 'icons'

### PR DESCRIPTION
**Pull request by mareklibra**
As an optimization, icons are provided in an array by
the updateIcons() action.

Useful for reload from LocalStorage.

### CODE REVIEW

File: src/reducers/icons.js [hunk 1]
```diff
@@ -1,4 +1,11 @@
-import Immutable, { Map } from 'immutable'
+import { Map } from 'immutable'
+
+function updateOrAddVm ({ state, payload: { icons } }) {
```
Comment (matobet) [inline src/reducers/icons.js]: @mareklibra are you really sure you want to call this method `updateOrAddVm`? Seems really icon-centric. If this is a result of a copy-paste from the VM equivalent probably that is a sign that we want to extract this to a separate helper called e.g. `mergeById` and call it from both reducers?

Comment (mareklibra) [reply src/reducers/icons.js]: ooops, thanks

File: src/components/VmDetail/index.js [hunk 2]
```diff
@@ -8,9 +8,36 @@ import { closeVmDetail } from '../../actions/vm'
 
 import VmIcon from '../VmIcon'
 import VmDisks from '../VmDisks'
-
+import Time from '../Time'
 import VmActions from '../VmActions'
 
+const LastMessage = ({ vmId, userMessages }) => {
+  const vmMessages = userMessages.get('records')
+    .filter(msg => { return (msg.failedAction && msg.failedAction.payload && msg.failedAction.payload.vmId === vmId) })
```
Comment (matobet) [inline src/components/VmDetail/index.js]: please make lambdas returning single value without the braces and `return`. Just `msg => expression` is sufficient.

File: src/components/VmDetail/index.js [hunk 3]
```diff
@@ -8,9 +8,36 @@ import { closeVmDetail } from '../../actions/vm'
...
+    .sort((msg1, msg2) => { return msg1.time - msg2.time })
```
Comment (matobet) [inline src/components/VmDetail/index.js]: same here

**Review by matobet — CHANGES_REQUESTED** [review]

**Review by matobet — APPROVED** [review]

**Review by matobet — APPROVED** [review]

**Comment by mareklibra** [comment]
Done

**Pull request merged**